### ISSUE TITLE
Inconsistency in HID usage

### ISSUE THREAD

**Issue by noseglasses** [keyboardio/KeyboardioHID#56]
`Multireport/Keyboard.cpp` and `MultiReport/Mouse.cpp` use the `HID` class to sent their reports. `SingleReport/SingleAbsoluteMouse.cpp` and `BootKeyboard/BootKeyboard.cpp` send them directly.

It would be nice for all devices to use the `HID` class for the sake of consistency and as this makes it possible to observe all HID reports via a common callback of the `HID` class (as it is already possible for keyboard and mouse reports). The latter is important for keyboard simulation and especially for simulation based integration testing where the generated HID reports and their precise timing are a major evaluation criterion.

**Comment by obra** [comment]
We inherited this from the original upstream library. If the actual USB
packets being sent are identical, I have no problem with this changing.
ᐧ

**Comment by noseglasses** [comment]
Absolute mouse and boot keyboard are individual instances of interface `PluggableUSBModule`. That's why they cannot use the singleton `HID`. That's something I overlooked before asking my previous question. 

Instead of changing the HID implementation I moved the HID reporting observer hook into a class of its own that is shared by all HID reporting classes. See https://github.com/keyboardio/KeyboardioHID/pull/57

In general, I spend some time trying to understand the upstream HID library, Keyboardio's fork and Kaleidoscope-HIDAdaptor-KeyboardioHID's usage of the provided interface. There are some points where I could use some clarification (because my knowledge about USB/HID is still pretty limited).

1) When are HID reports bundled to be come multi-reports and what are possible restrictions of this approach that require single reports?

2) Why does Kaleidoscope-HIDAdaptor-KeyboardioHID use the single-report version for absolute mouse reports although there is a multi-report version available?
